src/codeu/chat/server/BroadCastSystem.java

```diff
@@ -0,0 +1,148 @@
+package codeu.chat.server;
+
+import codeu.chat.common.*;
+import codeu.chat.util.Serializers;
+import codeu.chat.util.connections.Connection;
+import codeu.chat.util.Uuid;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.LinkedBlockingQueue;
+
+/**
+ * Created by rsharif on 3/17/17.
+ */
+
+
+public class BroadCastSystem {
+
+    private class ConversationMessageLink {
+
+        Uuid conversationUuid;
+        Message message;
+
+        ConversationMessageLink(Uuid conversationUuid, Message message) {
+            this.conversationUuid = conversationUuid;
+            this.message = message;
+        }
+
+    }
+
+    private ConcurrentHashMap<Integer, List<Connection>> conversationUsers;
```

Comment:
Can the key just be the Conversation object?  Or maybe Conversation.UUID instead of Conversation.UUID.id()?
I'm not sure why they wrote their own UUID class, but from scanning it, i think HashCode is created in such a way that different objects with the same content will hash to the same thing.  If you can hash key by conversation, that'll make the rest of the code much more intuitive.